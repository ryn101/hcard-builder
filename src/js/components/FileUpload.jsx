import React from 'react';
import PropTypes from 'prop-types';

function FileUpload({
    id,
    children,
    focusClass,
    onFileSelection,
}) {
    return (
        <span className="file-upload-wrapper">
            <input id={id} onChange={onFileSelection} type="file" accept="image/*" />
            <label htmlFor={id} className={focusClass}>{ children }</label>
        </span>
    );
}

FileUpload.propTypes = {
    id: PropTypes.string.isRequired,
    children: PropTypes.oneOfType([
        PropTypes.arrayOf(PropTypes.node),
        PropTypes.node,
    ]).isRequired,
    focusClass: PropTypes.string,
    onFileSelection: PropTypes.func.isRequired,
};

FileUpload.defaultProps = {
    focusClass: 'focusable',
};

export default FileUpload;
