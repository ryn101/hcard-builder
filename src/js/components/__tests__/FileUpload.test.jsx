import React from 'react';
import ReactDOM from 'react-dom';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import FileUpload from '../FileUpload';

configure({ adapter: new Adapter() });

it('renders the file upload component without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
        <FileUpload id="fileUploadId" onFileSelection={() => {}}>Upload</FileUpload>,
        div,
    );
    ReactDOM.unmountComponentAtNode(div);
});

describe('once rendered', () => {
    let fileUploadedNode;
    let fileSelected;
    let fileInput;
    const fileUploadId = 'fileUploadId';

    beforeEach(() => {
        fileSelected = jest.fn();
        fileUploadedNode = shallow(
            <FileUpload id={fileUploadId} onFileSelection={fileSelected}>Upload</FileUpload>,
        );
        fileInput = fileUploadedNode.find('input');
    });

    it('renders with state correctly', () => {
        expect(fileUploadedNode).toMatchSnapshot();
    });

    it('passes down the id attribute to the file input', () => {
        expect(fileInput.prop('id')).toEqual(fileUploadId);
    });

    it('calls on file selection method on file change', () => {
        fileInput.simulate('change', {});
        expect(fileSelected).toHaveBeenCalledTimes(1);
    });
});
