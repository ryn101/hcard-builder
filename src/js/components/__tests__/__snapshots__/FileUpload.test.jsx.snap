// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`once rendered renders with state correctly 1`] = `
<span
  className="file-upload-wrapper"
>
  <input
    accept="image/*"
    id="fileUploadId"
    onChange={[MockFunction]}
    type="file"
  />
  <label
    className="focusable"
    htmlFor="fileUploadId"
  >
    Upload
  </label>
</span>
`;
